src/wasm-interpreter.h

```diff
@@ -1490,6 +1490,15 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
     }
     const auto& value = flow.getSingleValue();
     NOTE_EVAL1(value);
+    if (curr->op == BrOnNull) {
+      // BrOnNull is the reverse of the others: it branches if null, and it does
+      // not flow out the existing value otherwise, rather it returns it as non-
+      // null.
```

Comment:
If I'm reading the proposal correctly, `br_on_null` does not propagate the value if it takes the branch. Also, it doesn't change the value if it doesn't take the branch, but rather refines the value's type. But the value's dynamic type doesn't change, so that doesn't matter here in the interpreter.

```suggestion
      // Unlike the others, BrOnNull does not propagate the value if it takes the
      // branch.
```